### PR TITLE
Add volumes field to DOMachine status

### PR DESCRIPTION
**Pull request by gottwald**
**What this PR does / why we need it**:

This will help when debugging because it's easier to find the associated volume that was provisioned with the droplet.

**Release note**:

```release-note
Add associated volumes to DOMachine status
```

### CODE REVIEW

File: api/v1alpha3/types.go [hunk 1]
```diff
@@ -127,6 +127,11 @@ type DOVPC struct {
 	VPCUUID string `json:"vpc_uuid,omitempty"`
 }
 
+// DOVolume defnies a DO Block Storage Volume.
```
Comment (timoreimann) [inline api/v1alpha3/types.go:130]: typo

(repeated elsewhere)

File: cloud/scope/machine.go [hunk 2]
```diff
@@ -143,6 +143,19 @@ func (m *MachineScope) SetProviderID(dropletID string) {
 	m.DOMachine.Spec.ProviderID = pointer.StringPtr(pid)
 }
 
+// SetVolumes sets the DOMachine volume IDs from doplet in status.
```
Comment (timoreimann) [inline cloud/scope/machine.go:146]: typo (doplet -> droplet)

Comment (gottwald) [reply cloud/scope/machine.go]: fixed

File: api/v1alpha3/types.go (hunk 1 above)
Comment (gottwald) [reply api/v1alpha3/types.go]: fixed

File: config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml [hunk 3]
```diff
@@ -182,6 +182,18 @@ spec:
               ready:
                 description: Ready is true when the provider resource is ready.
                 type: boolean
+              volumes:
+                description: Volumes contains the DigitalOcean droplet associated
+                  block storage volumes.
+                items:
+                  description: DOVolume defnies a DO Block Storage Volume.
```
Comment (MorrisLaw) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml:189]: typo: defnies -> defines

File: config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml [hunk 4]
```diff
@@ -524,6 +548,18 @@ spec:
...
```
Comment (MorrisLaw) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml:555]: same as typo above

**Review by MorrisLaw — CHANGES_REQUESTED** [review]
typo nits

File: api/v1alpha3/domachine_types.go [hunk 5]
```diff
@@ -57,6 +57,10 @@ type DOMachineStatus struct {
 	// Addresses contains the DigitalOcean droplet associated addresses.
 	Addresses []corev1.NodeAddress `json:"addresses,omitempty"`
 
+	// Volumes contains the DigitalOcean droplet associated block storage
```
Comment (MorrisLaw) [inline api/v1alpha3/domachine_types.go:60]: nit: I feel like this comment can fit in one line. It's not longer than the comment below it even when we don't wrap. This suggestion could be applied to other similar comments in this PR. Up to you though, this is not a blocker and purely preference on my part.

Comment (MorrisLaw) [inline api/v1alpha3/domachine_types.go:60]: nit: I feel like this comment can fit in one line. It's not longer than the comment below it even when we don't wrap. This suggestion could be applied to other similar comments in this PR. Up to you though, this is not a blocker and purely preference on my part.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml (hunk 3 above)
Comment (gottwald) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml]: Looks like I forgot to re-render the yamls after the fix. Thx.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml (hunk 4 above)
Comment (gottwald) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_domachines.yaml]: fixed

File: api/v1alpha3/domachine_types.go (hunk 5 above)
Comment (gottwald) [reply api/v1alpha3/domachine_types.go:60]: The majority of other comments even on that type seem to wrap at 80 chars max which is pretty common.
The comment below seems to be the exception.

Comment (MorrisLaw) [reply api/v1alpha3/domachine_types.go:60]: Ahh ok, fair enough

**Review by MorrisLaw — APPROVED** [review]
/lgtm

**Pull request merged**